src/App.tsx

```diff
@@ -1,223 +1,103 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+// eslint-disable-next-line object-curly-newline
+import React, {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { getTodos, createTodo } from './api/todos';
+import { filterTotos } from './api/filter';
+import { useAuthContext } from './components/Auth/useAuth';
+import {
+  ErrorNotification,
+  Header,
+  TodoList,
+  Footer,
+} from './components';
+import { Todo, Filter, Error } from './types';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const user = useContext(AuthContext);
+  const user = useAuthContext();
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [typeFilter, setTypeFilter] = useState(Filter.all);
+  const [error, setError] = useState('');
+  const [isHiddenErrorNote, setIsHiddenErrorNote] = useState(true);
+
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then((loadedTodos) => setTodos(loadedTodos))
+        .catch(() => {
+          setError(Error.loading);
+          setIsHiddenErrorNote(false);
+          setTimeout(() => {
+            setIsHiddenErrorNote(true);
+          }, 3000);
+        });
+    }
   }, []);
 
+  const handleAddToto = (titleTodo: string) => {
+    if (!user) {
+      return;
+    }
+
+    const todo: Todo = {
+      title: titleTodo,
+      userId: user?.id,
+      completed: false,
+      id: user?.id,
+    };
+
+    createTodo(todo)
+      .then((loadedTodos) => setTodos(loadedTodos))
+      .catch(() => {
+        setError(Error.add);
+        setIsHiddenErrorNote(false);
+        setTimeout(() => {
+          setIsHiddenErrorNote(true);
+        }, 3000);
```

Comment:
Duplicated code from 41-44 lines, it is better to move these commands into separated function